src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/DoNotDecreaseInheritedMemberVisibilityTests.cs

```diff
@@ -12,12 +10,295 @@ public class DoNotDecreaseInheritedMemberVisibilityTests : DiagnosticAnalyzerTes
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
+        }
+
+        [Fact]
+        public void DecreaseCSharpMemberVisibility()
+        {
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    private new int MyProp3;
+}", GetCSharpCA2222RuleNameResultAt(8, 21));
+
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    internal new int MyProp3 { get; set; }
+}", GetCSharpCA2222RuleNameResultAt(8, 22), GetCSharpCA2222RuleNameResultAt(8, 32), GetCSharpCA2222RuleNameResultAt(8, 37));
```

Comment:
Ok, got it. Thanks, @lgolding. It's awkward because there's really not a userful code fix to be made at the accessor level, so I think what I'll do for now is:

1. Only report a diagnostic for methods (or accessors) so that we have FxCop parity.
2. Not offer a code fix on any accessors that don't have an explicit visibility. In the future, if we update this to fire on the property, instead, then that code fix will be available.